ERCS/erc-7750.md

```diff
@@ -0,0 +1,103 @@
+---
+eip: 7750
+title: Decentralized Employment System
+description: An employment system that records employment history.
+author: James Savechives (@jamesavechives) <james.walstonn@gmail.com>
+discussions-to: https://ethereum-magicians.org/t/erc-7750-decentralized-employment-system-des/20724
+status: Draft
+type: Standards Track
+category: ERC
+created: 2024-08-04
+---
+
+## Abstract
+This ERC proposes a decentralized employment system (DES) built on the Ethereum blockchain. The system records employment history, allows users to found new companies, facilitates labor contracts, automates salary payments, incorporates a moderation mechanism for dispute resolution, and includes a review system for both companies and employees. All employment history is public, ensuring transparency and accountability in the employment process.
+
+It must be noted that this system comes into play after the company tests the employee, the employee passes these tests, and before hiring and signing the labor contract. The employee uses their employee token, which contains their employment history, and the company checks the history before finalizing the labor contract.
+
+## Motivation
+Traditional employment systems are centralized, opaque, and often lack trust. The DES aims to bring transparency, immutability, and trust to employment by leveraging blockchain technology. By recording employment history on-chain, enabling decentralized company creation, automating contract enforcement, and adding moderation for dispute resolution, the DES promotes a fairer and more transparent employment ecosystem. This system also streamlines the final steps of hiring, following testing phases, making the contract phase smoother and more secure.
+
+## Specification
+
+### Company Creation
+1. **Company Registration**: Users can register new companies on the blockchain. Each company is a unique entity with its own address.
+2. **Company Profile**: The company profile includes details like name, industry, and a list of registered employees.
+
+### Labor Contracts
+1. **Contract Creation**: Once the company has tested the employee and is ready to move forward with the final hiring stage, companies can create a labor contract by calling the `createContract(uint companyId, uint employeeTokenId, uint salary, uint duration, string responsibilities, string terminationConditions)` function. This function generates a unique `contractId` and stores it in a `contracts` mapping, associating each contract with a `companyId`.
```

Comment:
> stores it in a `contracts` mapping, associating each contract with a `companyId`

This is an implementation detail. ERCs should only specify the externally visible behaviour of a contract, not how the contract is implemented.